doc/logging/persistent_logger/persistent_loglevel.md

```diff
@@ -0,0 +1,217 @@
+1. Keep using the existing LOGLEVEL DB and make it persistent by adding loglevel_db.json file. 
+2. Move the LOGLEVEL DB content into the Config DB. Since the Config DB is already persistent, the log level will also be persistent to reboot. The log level will be saved when using the "config save" CLI command.
+
+We discarded this second approach to allow users to save the log level with deticated command and not exesiting command that will save other configuration too.
+
+### Making LOGLEVEL DB persistent by adding "loglevel_db.json" file
+
+ - We will add a new "log-level save" command to allow the user to save the log level.
+
+```
+admin@sonic:~$ log-level save
+```
+
+ 
+ - Similar to the config_db.json, we will add a new loglevel_db.json. The JSON file will be added to etc/sonic/loglevel_db.json.
+
+```json
+{
+  "LOGGER": {
+    "orchagent:orchagent": {
+      "LOGLEVEL": "INFO",
+      "LOGOUTPUT": "SYSLOG"
+    }
+  }
+}
+```
+
+- On "swssloglevel": We will set the log level to the LOGLEVEL DB, This is the behavior we have today, and it redmain as it is (with addition of set all components to default log level).
+- On "log-level save": We will copy the LOGLEVEL DB content into the loglevel_db.json.
+- On startup: we will load the loglevel_db.json file into the LOGLEVEL DB.
+
+
+## 3.2 Persistent logger flow
+
+- Each component has a singleton Logger object with a log level property and listener thread.
+- On startup:
+  - The loglevel_db.json file is loaded into the LOGLEVEL DB during the init of database docker.
+  - The loglevel property is set accordingly to the loglevel value on the LOGLEVEL DB.
+- When a component writes a log message, the Logger writes the message only if the loglevel of the message is above the current loglevel property.
+- When the user wants to set a new log level to a component, he uses the "swssloglevel" CLI command. The "swssloglevel" script sets the new verbosity to the LOGLEVEL DB (database #3).
+- The LOGLEVEL DB change triggers an event, which is caught by the listener thread.
+- The listener thread changes the loglevel property of the Logger.
+- The user can use the "log-level save" command to save the current loglevel and make it persistent. It will copy the LOGLEVEL DB content into the loglevel_db.json.
+In addition to the log level, the LOGLEVEL DB contains the log output file. After "log-level save" the log output will be persistent to reboot too.
+
+### 3.2.1 Return to default log level
+
+  To return to the default log level, the user will remove the loglevel_db.json and reboot the switch.
+
+
+
+
+![persistent logger flow](/doc/logging/persistent_logger/persistent_logger.png)
+
+
+
+
+
+# 4 Flows
+
+## 4.1 Logger init flow
+
+When the system startup and the Database container initialize, we will load the loglevel_db.json into the LOGLEVEL DB (similar to the config_db.json). If the loglevel.json file is deleted, the system will generate a new loglevel_db.json file with default log level values.
```

Comment:
If the loglevel.json file is deleted --> If the loglevel.json file is not exist
and if it is not exist, the system will not generate loglevel_db.json automatically. only after the user run log-level save

Reply:
Done